### PR TITLE
Improved actor match on freeones plugin

### PR DESCRIPTION
**Pull request by arcadianCdr**
I have a large collection and many actress were incorrectly parsed by the current freeones plugin. To correctly match as best as possible the queried actor, I added the sort by rank parameter to the actor search query.

I also added a test class for actress "Eva Angelina" as they are lots of Eva and lots of Angelina, it is a good test case to make sure the first match is actually "Eva Angelina" and not another wrong match (the current version of the plugin incorrectly maps her to the russian actress Evangelina Wish...)

I had also to modify the test case for Gia DiMarco (that is an alias for Haley Wilde). My improved match correctly matches to Gia DiMarca where the current plugin matches "Gianna Di Marco", which is incorrect. The Gia DiMarco test case is now adapted to match the actual Gia DiMarco (Haley Wilde) data.

### CODE REVIEW

**Comment by leadwolf** [comment]
The files need to be formatted with Prettier. If you use VSCode, just install the extension.  
It works, but how did you even find the "rank.currentRank" param ? On the site, I only see "relevance".

**Comment by arcadianCdr** [comment]
Thanks for the tip on Prettier. Reformatting done and committed.
For the sort params, I just use the site from the home page and there you have all the sort orders available. Params on the main search are also valid on the /partial/subject target.
Sort options:
- Age
- Babe rank
- Cup Size
- Follower Count
- Height
- Hip
- Latest
- Name
- Profile Completed
- Rank change
- Relevance
- Views
- Waist
- Weight

**Comment by boi123212321** [comment]
Test seems to be failing:

```
1) freeones
       Search 'Gia DiMarco' without measurements:

      AssertionError: expected { Object (hair color, eye color, ...) } to deeply equal { Object (hair color, eye color, ...) }
      + expected - actual

         "hair color": "Brown"
         "height": 163
         "piercings": "Left Nostril; Clitoris"
         "sex": "Female"
      -  "tattoos": "Small Stars Behind Right Ear; 2 Red Flowers With Leaves Upper Back; Large Design Of Flames, Wings And 6 Red Roses Lower Back; \"La Bella Vita\" Left Side Of Torso; Traces Of Removed Floral Right Hip; Cherries, Hearts And Stars Above Of Pubes; Nautical Motif"
      +  "tattoos": "Small·Stars·Behind·Right·Ear;·2·Red·Flowers·With·Leaves·Upper·Back;·Large·Design·Of·Flames,·Wings·And·6·Red·Roses·Lower·Back;·\"La·Bella·Vita\"·Left·Side·Of·Torso;·Traces·Of·Removed·Floral·Right·Hip;·Cherries,·Hearts·And·Stars·Above·Of·Pubes;·Nautical·Motif"
         "weight": 50
         "zodiac": "Scorpio"
       }
      
      at /home/runner/work/porn-vault-plugins/porn-vault-plugins/plugins/freeones/test/gia_dimarco.spec.js:63:35
      at Generator.next (<anonymous>)
      at fulfilled (plugins/freeones/test/gia_dimarco.spec.js:5:58)
      at processTicksAndRejections (internal/process/task_queues.js:93:5)
```

**Comment by arcadianCdr** [comment]
Indeed. It seems I messed-up something when reformatting with Prettier. That is now fixed. I rerun all tests locally and GitHub also confirms all are now passed.

**Review by leadwolf — APPROVED** [review]
OK. I think it's good. Why not just sort by relevance though ? Depending on how the search works, maybe someone could search for a low rank actor, but get a wrong match because someone of a similar name but with higher rank.

**Comment by boi123212321** [comment]
Maybe use an argument, like:
```
s: args.sort || "rank.currentRank"
```

**Comment by arcadianCdr** [comment]
I tested with hundreds of actors and indeed, 'relevance' gives better results. I went with this as default and defined it as parameter as suggested by boi. I'm a bit puzzled as to why Git's checks are failing at build while everything works fine for me locally...

**Comment by leadwolf** [comment]
Maybe because I updated the context and logging, but your branch has the old context. Merge the upstream and update your branch

**Comment by arcadianCdr** [comment]
Indeed. All should be OK now. Thanks for your support. As you might have guessed, this is my first github experience...

File: plugins/freeones/info.json [hunk 1]
```diff
@@ -26,6 +26,13 @@
       "default": [],
       "description": "Array of data fields to omit (for values see whitelist)"
     },
+    {
+      "name": "searchResultsSort",
+      "type": "String",
+      "required": false,
+      "default": "relevance",
+      "description": "Specify the search result sort order key to use. Advanced setting: use only if you knwo what you are doing. Possible values are 'relevance', 'rank.currentRank', 'followerCount', 'profileCompleted' or 'views'"
```
Comment (leadwolf) [inline plugins/freeones/info.json:34]: tiny typo
```suggestion
      "description": "Specify the search result sort order key to use. Advanced setting: use only if you know what you are doing. Possible values are 'relevance', 'rank.currentRank', 'followerCount', 'profileCompleted' or 'views'"
```

**Review by leadwolf — APPROVED** [review]
Good to go

**Comment by arcadianCdr** [comment]
Done.

**Comment by arcadianCdr** [comment]
I also adapted the AdultEmpire description scraping so it gets the full description instead of the partial one it gets now. This avoids getting the description truncated with a "Read more...".

**Comment by leadwolf** [comment]
> I also adapted the AdultEmpire description scraping so it gets the full description instead of the partial one it gets now. This avoids getting the description truncated with a "Read more...".

Could you add a test for a scene that would've had "Read more", but will now be the real decscription ?

**Comment by arcadianCdr** [comment]
@leadwolf I removed the AdultEmpire changes from this pull request on Freeones so you can already approve it (as this part is now complete). 

I will create another pull request for the AdultEmpire improvements I'm working on: full description, thumbnail image in addition to avatar and hero, rating,... 

The full description contains paragraphs as newlines in most cases and will also require a small change in porn-vault's app CSS to enable pre-formatted text (something like "white-space: pre;"). Having a long description in one block is not a nice visual experience...

**Pull request merged**